Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPITestCocoa.mm

```diff
@@ -438,8 +438,17 @@
 
     auto testComplete = [this, protectedThis = Ref { *this }, test](JSValue *result, JSValue *error) {
         if (error || m_hitAssertion) {
-            JSValue *errorMessageValue = error.isObject && [error hasProperty:@"message"] ? error[@"message"] : error;
-            WebProcess::singleton().send(Messages::WebExtensionController::TestFinished(test.testName, false, adoptNS([[NSString alloc] initWithFormat:@"Promise rejected with an error: %@", debugString(errorMessageValue)]).get(), test.location.first, test.location.second), test.webExtensionControllerIdentifier);
+            NSString *errorMessage;
+            if (error) {
+                JSValue *errorMessageValue = error.isObject && [error hasProperty:@"message"] ? error[@"message"] : error;
+                errorMessage = adoptNS([[NSString alloc] initWithFormat:@"Promise rejected with an error: %@", debugString(errorMessageValue)]).get();
+            } else if (!m_assertionMessage.isNull())
+                errorMessage = adoptNS([[NSString alloc] initWithFormat:@"Promise rejected with an error: %@", m_assertionMessage.createNSString().get()]).get();
+            else
+                errorMessage = @"Promise rejected without an error";
```

Comment:
I'm wondering if we should just set the message as-is and not prefix with "Promise rejected with an error:". Seems odd to bake that in here.

Reply:
i kind of like the prefix. at least in terms of readability of the error in WPT. this is what it'd look like
`safari-web-extension://D39F4C09-2D13-406E-AAB8-75C89823321C/background.js:10. Promise rejected with an error: expected different tab id`